src/lib/core/data-table/data-table.spec.ts

```diff
@@ -264,6 +377,41 @@ class SimpleCdkTableApp {
 
 @Component({
   template: `
+    <cdk-table [dataSource]="dataSource" [trackBy]="trackBy">
+      <ng-container cdkColumnDef="column_a">
+        <cdk-header-cell *cdkHeaderCellDef> Column A</cdk-header-cell>
+        <cdk-cell *cdkCellDef="let row"> {{row.a}}</cdk-cell>
+      </ng-container>
+
+      <ng-container cdkColumnDef="column_b">
+        <cdk-header-cell *cdkHeaderCellDef> Column B</cdk-header-cell>
+        <cdk-cell *cdkCellDef="let row"> {{row.b}}</cdk-cell>
+      </ng-container>
+
+      <cdk-header-row *cdkHeaderRowDef="columnsToRender"></cdk-header-row>
+      <cdk-row *cdkRowDef="let row; columns: columnsToRender"></cdk-row>
+    </cdk-table>
+  `
+})
+class TrackByCdkTableApp {
+  static TRACK_BY: 'reference' | 'propertyA' | 'index' = 'reference';
```

Comment:
It cannot be a component property because the `trackBy` function does not run in the context of the component (cannot use `this`). Fixed the caps case, thought I saw a lint error regarding the format, but I must have confused it with a different property.

Static because the `trackBy` function does not have `this` as `TrackByCdkTableApp` so it cannot be